include/tins/memory_helpers.h

```diff
@@ -141,11 +150,40 @@ class InputMemoryStream {
     void read(HWAddress<6>& address);
     void read(IPv4Address& address);
     void read(IPv6Address& address);
-private:
+protected:
```

Comment:
I originally initialized these two members in the child class constructor, which requires that these members be protected instead of private. I've resolved this issue by calling the parent class constructor in the child class constructor.